File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        
        # return length of longest consecutive sequence of elements
        # valid sequence is when each next element is EXACTLY 1 greater than previous (does not have to be adjacent elements)
        # must be O(n) time, 

        # Create a numSet into a set
        numSet = set(nums)
        maxCount = 0
        
        # Iterate through the set
        for num in numSet:
            count = 0

            # Sequence start number is only valid, if number - 1 does not exist in the dictionary
            if (num - 1) not in numSet:
                count += 1

                # while number + 1, exists, increment count
                while (num + 1) in numSet:
                    count += 1
                    num += 1
        
            # established max count
            maxCount = max(maxCount, count)
       
        return maxCount 

File: Data Structures & Algorithms/duplicate-integer/submission-2.cs
public class Solution {
    public bool hasDuplicate(int[] nums) {
        // Create a dictionary to store the nums
        HashSet<int> seenNums = new HashSet<int>();
        
        // Iterate over the nums, 
        foreach (int num in nums) {
            Console.WriteLine(num);
            // If HashSet not empty and num in HashSet, return true
            if (seenNums.Count > 0 && seenNums.Contains(num)) {
                return true;
            } else {
                seenNums.Add(num);
            }

            // Else add to the Hashset
            seenNums.Add(num);
        }

        // When there are no duplicates return false
        return false; 
    }
}
